src/Microsoft.VisualStudio.Services.Agent/JobServerQueue.cs

```diff
@@ -362,26 +287,15 @@ public void QueueTimelineRecordUpdate(Guid timelineId, TimelineRecord timelineRe
         {
             while (!_jobCompletionSource.Task.IsCompleted || runOnce)
             {
-                if (!runOnce)
-                {
-                    bool forceProcess = await _fileUploadQueueSemaphore.WaitAsync(_delayForFileUploadDequeue);
-                    if (forceProcess)
-                    {
-                        Trace.Verbose("Process file upload queue since the force process signal been fired.");
-                    }
-                    else
-                    {
-                        Trace.Verbose("Process file upload queue since the time delay has met.");
-                    }
-                }
-
                 List<UploadFileInfo> filesToUpload = new List<UploadFileInfo>();
-                lock (_fileUploadQueueLock)
+                UploadFileInfo dequeueFile;
+                while (_fileUploadQueue.TryDequeue(out dequeueFile))
                 {
-                    while (_fileUploadQueue.Count > 0)
+                    filesToUpload.Add(dequeueFile);
+                    // process at most 5 file upload.
+                    if (filesToUpload.Count > 5)
```

Comment:
...although I guess we already have at least 3 other communication channels already going between the client/server.